- Select dataset
- Perform data preparation and cleaning using Pandas & Numpy
- Perform exploratory analysis & visualization using Matplotlib and Seaborn
- Ask and Answer questions about the data in a Jupyter Notebook
- Summarize your inference and write a conclusion
- Document, publish and present your Jupyter Notebook Online
- Create interactive Dashboard 

# American Values Data Analysis

TODO- talk about EDA

TODO- talk about the dataset (source, what it contains, how it will be useful)

- Source: Five Thirty Eight
- Can be useful to understand American values how 

# Data Preparation and Cleaning

- Load the file using pandas
- Look at some information about the file about the data and columns
- Fix any missing or incorrect values


## Load the file using pandas

In [2]:
import numpy as np
import pandas as pd

df = pd.read_csv('nonvoters_data.csv')


df.head()

RespId  weight  Q1  Q2_1  Q2_2  Q2_3  Q2_4  Q2_5  Q2_6  Q2_7  ...  Q30  \
0  470001  0.7516   1     1     1     2     4     1     4     2  ...    2   
1  470002  1.0267   1     1     2     2     3     1     1     2  ...    3   
2  470003  1.0844   1     1     1     2     2     1     1     2  ...    2   
3  470007  0.6817   1     1     1     1     3     1     1     1  ...    2   
4  480008  0.9910   1     1     1    -1     1     1     1     1  ...    1   

   Q31  Q32  Q33  ppage                 educ   race  gender     income_cat  \
0  NaN  1.0  NaN     73              College  White  Female       $75-125k   
1  NaN  NaN  1.0     90              College  White  Female  $125k or more   
2  NaN  2.0  NaN     53              College  White    Male  $125k or more   
3  NaN  1.0  NaN     58         Some college  Black  Female        $40-75k   
4 -1.0  NaN  NaN     81  High school or less  White    Male        $40-75k   

   voter_category  
0          always  
1          always  
2        sporadic  
3        sporadic  
4          always  

[5 rows x 119 columns]

## Look at information about the data and columns

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5836 entries, 0 to 5835
Columns: 119 entries, RespId to voter_category
dtypes: float64(23), int64(91), object(5)
memory usage: 5.3+ MB


By looking at the info from our dataframe, we learn that we have 119 columns in our dataset and 5,836 rows. 

There's too many columns to see the full number and so, to find out information about the columns, [we refer to the codebook for the dataset](https://github.com/fivethirtyeight/data/blob/master/non-voters/nonvoters_codebook.pdf).

This analysis will focus on questions two and three in the study and so we'll filter the dataframe to only include the pertinent columns.


In [4]:
df.describe()

RespId       weight      Q1         Q2_1         Q2_2  \
count    5836.000000  5836.000000  5836.0  5836.000000  5836.000000   
mean   474653.997772     0.991023     1.0     1.246402     1.705106   
std      3628.475677     0.345022     0.0     0.660253     0.866346   
min    470001.000000     0.229800     1.0    -1.000000    -1.000000   
25%    472069.750000     0.793150     1.0     1.000000     1.000000   
50%    474152.000000     0.967600     1.0     1.000000     2.000000   
75%    476217.500000     1.169600     1.0     1.000000     2.000000   
max    488325.000000     3.038600     1.0     4.000000     4.000000   

              Q2_3         Q2_4         Q2_5         Q2_6         Q2_7  ...  \
count  5836.000000  5836.000000  5836.000000  5836.000000  5836.000000  ...   
mean      1.638280     2.175977     1.277245     1.805517     1.491604  ...   
std       0.765741     1.091391     0.626386     1.011524     0.808120  ...   
min      -1.000000    -1.000000    -1.000000    -1.000000    -1.000000  ...   
25%       1.000000     1.000000     1.000000     1.000000     1.000000  ...   
50%       2.000000     2.000000     1.000000     1.000000     1.000000  ...   
75%       2.000000     3.000000     1.000000     2.000000     2.000000  ...   
max       4.000000     4.000000     4.000000     4.000000     4.000000  ...   

             Q29_6        Q29_7        Q29_8        Q29_9       Q29_10  \
count  1342.000000  1342.000000  1342.000000  1342.000000  1342.000000   
mean     -0.926975    -0.758569    -0.697466    -0.810730    -0.700447   
std       0.375264     0.651835     0.716885     0.585638     0.713970   
min      -1.000000    -1.000000    -1.000000    -1.000000    -1.000000   
25%      -1.000000    -1.000000    -1.000000    -1.000000    -1.000000   
50%      -1.000000    -1.000000    -1.000000    -1.000000    -1.000000   
75%      -1.000000    -1.000000    -1.000000    -1.000000    -1.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

               Q30          Q31          Q32          Q33        ppage  
count  5836.000000  1592.000000  2002.000000  2242.000000  5836.000000  
mean      2.325051     1.364950     1.365634     1.220339    51.693797  
std       1.259642     0.519249     0.497046     0.958569    17.071561  
min      -1.000000    -1.000000    -1.000000    -1.000000    22.000000  
25%       1.000000     1.000000     1.000000     1.000000    36.000000  
50%       2.000000     1.000000     1.000000     1.000000    54.000000  
75%       3.000000     2.000000     2.000000     2.000000    65.000000  
max       5.000000     2.000000     2.000000     2.000000    94.000000  

[8 rows x 114 columns]

In [5]:
# Are we missing any values? We'll create a for loop that only shows us the values for columns with more than one missing value


df.isna().sum().sort_values()

RespId       0
Q19_2        0
Q19_1        0
Q18_10       0
Q18_9        0
          ... 
Q29_3     4494
Q29_2     4494
Q29_9     4494
Q29_4     4494
Q22       5350
Length: 119, dtype: int64

We are missing values but they're for specific response questions, we'll wait to check for missing values again when we have our final, processed dataframe. 

Now we'll rename those columns and then create a new dataframe with the new column names and the socioeconomic columns.

In [6]:
# Create dictionary with corresponding keys

column_dict = {"Q2_1":"Voting",
        "Q2_2": "Jury",
        "Q2_3": "Govt_Politics",
        "Q2_4": "American_Flag",
        "Q2_5": "US_Census",
        "Q2_6": "Pledge",
        "Q2_7":"Military Support",
        "Q2_8": "Respecting_Others",
        "Q2_9":"Believing in God",
        "Q2_10":"Protesting", 
        "Q3_1": "Racism_US",
        "Q3_2": "Racism_Policing",
        "Q3_3":"Society_Whole",
        "Q3_4":"Mainstream_Media",
        "Q3_5":"Traditional_Parties",
        "Q3_5":"Way_People_Talk"}

# Create list of dictionary 
keylist = list(column_dict.keys())

# Create two lists that we'll use to restructure the survey data
question_two = []
question_three = []
for key in keylist:
    if "Q2" in key:
        question_two.append(key)
    else:
        question_three.append(key)

# Likert scale data (make agreeable

# Create two lists for questions two and three, restructure data that way

# Age (bins)

# make the bins by quartile
df


RespId  weight  Q1  Q2_1  Q2_2  Q2_3  Q2_4  Q2_5  Q2_6  Q2_7  ...  Q30  \
0     470001  0.7516   1     1     1     2     4     1     4     2  ...    2   
1     470002  1.0267   1     1     2     2     3     1     1     2  ...    3   
2     470003  1.0844   1     1     1     2     2     1     1     2  ...    2   
3     470007  0.6817   1     1     1     1     3     1     1     1  ...    2   
4     480008  0.9910   1     1     1    -1     1     1     1     1  ...    1   
...      ...     ...  ..   ...   ...   ...   ...   ...   ...   ...  ...  ...   
5831  477662  1.1916   1     1     3     1     3     1     2     2  ...    2   
5832  477663  1.4623   1     1     1     1     2     1     2     1  ...    2   
5833  488322  0.9252   1     1     2     1     3     1     1     2  ...    2   
5834  488325  2.6311   1     2     2     2     2     2     2     2  ...    3   
5835  477666  1.6218   1     1     3     2     3     1     1     2  ...    5   

      Q31  Q32  Q33  ppage                 educ         race  gender  \
0     NaN  1.0  NaN     73              College        White  Female   
1     NaN  NaN  1.0     90              College        White  Female   
2     NaN  2.0  NaN     53              College        White    Male   
3     NaN  1.0  NaN     58         Some college        Black  Female   
4    -1.0  NaN  NaN     81  High school or less        White    Male   
...   ...  ...  ...    ...                  ...          ...     ...   
5831  NaN  1.0  NaN     27         Some college     Hispanic    Male   
5832  NaN  2.0  NaN     59  High school or less        White  Female   
5833  NaN  1.0  NaN     51              College  Other/Mixed    Male   
5834  NaN  NaN  1.0     22  High school or less        Black  Female   
5835  NaN  NaN  2.0     22  High school or less        Black  Female   

          income_cat  voter_category  
0           $75-125k          always  
1      $125k or more          always  
2      $125k or more        sporadic  
3            $40-75k        sporadic  
4            $40-75k          always  
...              ...             ...  
5831         $40-75k          always  
5832   $125k or more    rarely/never  
5833   $125k or more        sporadic  
5834  Less than $40k          always  
5835  Less than $40k          always  

[5836 rows x 119 columns]

2. In your view, how important are each of the following to being a good American?

   1. Voting in elections
   2. Serving on a jury
   3. Following what happens in government and politics
   4. Displaying the American flag
   5. Participating in the U.S. Census every ten years
   6. Knowing the Pledge of Allegiance
   7. Supporting the military
   8. Respecting the opinions of those who disagree with you
   9. Believing in God
   10. Protesting if you believe government actions are wrong
    
Responses
1. Very important
2. Somewhat important
3. Not so important
4. Not at all important


3. How much do you agree or disagree with the following statements?

   1. Systemic racism is a problem in the United States.
   2. Systemic racism in policing is a bigger problem than violence and vandalism
   in protests.
   1. Society as a whole has become too soft and feminine.
   2. The mainstream media is more interested in making money than telling the truth.
   3. Traditional parties and politicians don’t care about people like me.
   4. The way people talk needs to change with the times to be more sensitive to people with different backgrounds.

Responses
1. Strongly agree
2. Somewhat agree
3. Somewhat disagree
4. Strongly disagree

In [7]:
# Create list of original column names
column_list = list(df.columns)

question_two_columns = []
question_three_columns = []

for key in column_list:
    if "Q2_" in key:
        question_two_columns.append(key)
    elif "Q3_" in key:
        question_three_columns.append(key)


In [8]:
# Restructure Likert data for question two

question_two_new_values = ["Somewhat important","Not so important","Not at all important"]

 
# Question one

column_name = "Q2_1"



conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Very important"))

# Question two

column_name = "Q2_2"

conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Very important"))

# Question three

column_name = "Q2_3"


conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Very important"))

# Question four

column_name = "Q2_4"


conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Very important"))

# Question five

column_name = "Q2_5"


conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Very important"))

# Question six

column_name = "Q2_6"


conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Very important"))

# Question seven

column_name = "Q2_7"


conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Very important"))

# Question eight

column_name = "Q2_8"


conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Very important"))

# Question nine

column_name = "Q2_9"


conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Very important"))

# Question ten

column_name = "Q2_10"


conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Very important"))


In [9]:
column_dict = {"Q2_1":"Voting",
        "Q2_2": "Jury",
        "Q2_3": "Govt_Politics",
        "Q2_4": "American_Flag",
        "Q2_5": "US_Census",
        "Q2_6": "Pledge",
        "Q2_7":"Military Support",
        "Q2_8": "Respecting_Others",
        "Q2_9":"Believing in God",
        "Q2_10":"Protesting"}


df = df.rename(columns=column_dict)

df


RespId  weight  Q1              Voting                Jury  \
0     470001  0.7516   1      Very important      Very important   
1     470002  1.0267   1      Very important  Somewhat important   
2     470003  1.0844   1      Very important      Very important   
3     470007  0.6817   1      Very important      Very important   
4     480008  0.9910   1      Very important      Very important   
...      ...     ...  ..                 ...                 ...   
5831  477662  1.1916   1      Very important    Not so important   
5832  477663  1.4623   1      Very important      Very important   
5833  488322  0.9252   1      Very important  Somewhat important   
5834  488325  2.6311   1  Somewhat important  Somewhat important   
5835  477666  1.6218   1      Very important    Not so important   

           Govt_Politics         American_Flag           US_Census  \
0     Somewhat important  Not at all important      Very important   
1     Somewhat important      Not so important      Very important   
2     Somewhat important    Somewhat important      Very important   
3         Very important      Not so important      Very important   
4         Very important        Very important      Very important   
...                  ...                   ...                 ...   
5831      Very important      Not so important      Very important   
5832      Very important    Somewhat important      Very important   
5833      Very important      Not so important      Very important   
5834  Somewhat important    Somewhat important  Somewhat important   
5835  Somewhat important      Not so important      Very important   

                    Pledge    Military Support  ... Q30  Q31  Q32  Q33  ppage  \
0     Not at all important  Somewhat important  ...   2  NaN  1.0  NaN     73   
1           Very important  Somewhat important  ...   3  NaN  NaN  1.0     90   
2           Very important  Somewhat important  ...   2  NaN  2.0  NaN     53   
3           Very important      Very important  ...   2  NaN  1.0  NaN     58   
4           Very important      Very important  ...   1 -1.0  NaN  NaN     81   
...                    ...                 ...  ...  ..  ...  ...  ...    ...   
5831    Somewhat important  Somewhat important  ...   2  NaN  1.0  NaN     27   
5832    Somewhat important      Very important  ...   2  NaN  2.0  NaN     59   
5833        Very important  Somewhat important  ...   2  NaN  1.0  NaN     51   
5834    Somewhat important  Somewhat important  ...   3  NaN  NaN  1.0     22   
5835        Very important  Somewhat important  ...   5  NaN  NaN  2.0     22   

                     educ         race  gender      income_cat  voter_category  
0                 College        White  Female        $75-125k          always  
1                 College        White  Female   $125k or more          always  
2                 College        White    Male   $125k or more        sporadic  
3            Some college        Black  Female         $40-75k        sporadic  
4     High school or less        White    Male         $40-75k          always  
...                   ...          ...     ...             ...             ...  
5831         Some college     Hispanic    Male         $40-75k          always  
5832  High school or less        White  Female   $125k or more    rarely/never  
5833              College  Other/Mixed    Male   $125k or more        sporadic  
5834  High school or less        Black  Female  Less than $40k          always  
5835  High school or less        Black  Female  Less than $40k          always  

[5836 rows x 119 columns]

In [10]:
# Restructure Likert data for question three

question_three_new_values = ["Somewhat agree", "Somewhat disagree", "Strongly disagree"]

# Question one

column_name = "Q3_1"

conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Strong agree"))

# Question two

column_name = "Q3_2"

conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Strong agree"))

# Question three

column_name = "Q3_3"

conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Strong agree"))

# Question four

column_name = "Q3_4"

conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Strong agree"))

# Question five

column_name = "Q3_5"

conditions = [
      np.logical_and(df[column_name].gt(1), np.less_equal(df[column_name], 2)),
      np.logical_and(df[column_name].gt(2), np.less_equal(df[column_name], 3)),
      np.logical_and(df[column_name].gt(3), np.less_equal(df[column_name], 4)),]

df[column_name] = pd.Series(np.select(conditions, question_two_new_values, "Strong agree"))


In [19]:
# Rename the columns

column_dict = {"Q2_1":"Voting",
        "Q2_2": "Jury",
        "Q2_3": "Govt_Politics",
        "Q2_4": "American_Flag",
        "Q2_5": "US_Census",
        "Q2_6": "Pledge",
        "Q2_7":"Military Support",
        "Q2_8": "Respecting_Others",
        "Q2_9":"Believing in God",
        "Q2_10":"Protesting", 
        "Q3_1": "Racism_US",
        "Q3_2": "Racism_Policing",
        "Q3_3":"Society_Whole",
        "Q3_4":"Mainstream_Media",
        "Q3_5":"Traditional_Parties",
        "Q3_5":"Way_People_Talk"}


df = df.rename(columns=column_dict)


In [20]:
# Create a subset of our data


df = df.copy().loc[:,["Voting",
                "Jury",
                "Govt_Politics",
                "American_Flag",
                "US_Census",
                "Pledge",
                "Military Support",
                "Respecting_Others",
                "Believing in God",
                "Protesting",
                "Racism_US",
                "Racism_Policing",
                "Society_Whole",
                "Mainstream_Media",
                "Way_People_Talk",
                "ppage",
                "educ",
                "race",
                "gender",
                "income_cat",
                "voter_category"]]

In [ ]:
# Checking for missing values for our dataframe
df.isna().sum().sort_values()

Voting               0
gender               0
race                 0
educ                 0
ppage                0
Way_People_Talk      0
Mainstream_Media     0
Society_Whole        0
Racism_Policing      0
income_cat           0
Racism_US            0
Believing in God     0
Respecting_Others    0
Military Support     0
Pledge               0
US_Census            0
American_Flag        0
Govt_Politics        0
Jury                 0
Protesting           0
voter_category       0
dtype: int64

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5836 entries, 0 to 5835
Data columns (total 21 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Voting             5836 non-null   object
 1   Jury               5836 non-null   object
 2   Govt_Politics      5836 non-null   object
 3   American_Flag      5836 non-null   object
 4   US_Census          5836 non-null   object
 5   Pledge             5836 non-null   object
 6   Military Support   5836 non-null   object
 7   Respecting_Others  5836 non-null   object
 8   Believing in God   5836 non-null   object
 9   Protesting         5836 non-null   object
 10  Racism_US          5836 non-null   object
 11  Racism_Policing    5836 non-null   object
 12  Society_Whole      5836 non-null   object
 13  Mainstream_Media   5836 non-null   object
 14  Way_People_Talk    5836 non-null   object
 15  ppage              5836 non-null   int64 
 16  educ               5836 non-null   object


We can see that we have fifteen questions for our dataset (now restructured as string data). 

The other six columns represent socioeconomic data points. Aside from age, these are all object data types (or categorical data).


Last but not least, we'll bin our age data so that it's easier to visualize.

In [ ]:
df["ppage"].describe()

count    5836.000000
mean       51.693797
std        17.071561
min        22.000000
25%        36.000000
50%        54.000000
75%        65.000000
max        94.000000
Name: ppage, dtype: float64

In [ ]:
# Use quartile values to create bins

age_bins = ["22-36","37-54","55-65","66-94"]

df['age_binned'] = pd.qcut(df['ppage'],
                              q=[0,0.25, 0.5, 0.75, 1],
                              labels=age_bins)


Voting                Jury       Govt_Politics  \
343         Very important      Very important      Very important   
515   Not at all important    Not so important    Not so important   
829     Somewhat important    Not so important    Not so important   
989         Very important  Somewhat important  Somewhat important   
1022        Very important      Very important  Somewhat important   
...                    ...                 ...                 ...   
5746        Very important      Very important      Very important   
5764        Very important      Very important  Somewhat important   
5768        Very important  Somewhat important  Somewhat important   
5790        Very important  Somewhat important      Very important   
5828        Very important      Very important  Somewhat important   

             American_Flag           US_Census                Pledge  \
343     Somewhat important      Very important    Somewhat important   
515   Not at all important  Somewhat important  Not at all important   
829     Somewhat important    Not so important      Not so important   
989   Not at all important  Somewhat important      Not so important   
1022  Not at all important      Very important      Not so important   
...                    ...                 ...                   ...   
5746        Very important      Very important        Very important   
5764      Not so important  Somewhat important    Somewhat important   
5768      Not so important  Somewhat important  Not at all important   
5790    Somewhat important      Very important        Very important   
5828  Not at all important  Somewhat important  Not at all important   

          Military Support     Respecting_Others      Believing in God  \
343         Very important        Very important      Not so important   
515   Not at all important        Very important  Not at all important   
829       Not so important    Somewhat important    Somewhat important   
989     Somewhat important    Somewhat important        Very important   
1022      Not so important        Very important  Not at all important   
...                    ...                   ...                   ...   
5746        Very important        Very important      Not so important   
5764        Very important        Very important        Very important   
5768    Somewhat important  Not at all important  Not at all important   
5790        Very important    Somewhat important        Very important   
5828    Somewhat important        Very important  Not at all important   

              Protesting  ...         Society_Whole      Mainstream_Media  \
343   Somewhat important  ...      Not so important    Somewhat important   
515       Very important  ...  Not at all important          Strong agree   
829   Somewhat important  ...      Not so important    Somewhat important   
989   Somewhat important  ...  Not at all important    Somewhat important   
1022      Very important  ...  Not at all important          Strong agree   
...                  ...  ...                   ...                   ...   
5746      Very important  ...          Strong agree          Strong agree   
5764    Not so important  ...  Not at all important          Strong agree   
5768      Very important  ...          Strong agree  Not at all important   
5790      Very important  ...      Not so important  Not at all important   
5828      Very important  ...    Somewhat important          Strong agree   

           Way_People_Talk ppage                 educ         race  gender  \
343     Somewhat important    37              College        White    Male   
515           Strong agree    37              College  Other/Mixed  Female   
829     Somewhat important    37              College  Other/Mixed    Male   
989     Somewhat important    37              College        White    Male   
1022    Somewhat important    37              College        White    Male   
...                    ...   ...

## Fix any missing or incorrect values


In [ ]:
# Create list of survey data columns - need to use to show that there's data types that will need to be restructured

survey_data_columns = []
for column in df.columns:
    if df[column].dtypes == "int64" and column != 'ppage':
        survey_data_columns.append(column)

In [ ]:
# Filter for just response data- might need to add this at the beginning (before column names are changed to more clear )




# Perform exploratory analysis & visualization using Matplotlib and Seaborn

# Ask and Answer questions about the data in a Jupyter Notebook

- Do people of certain backgrounds tend to think differently American values?
- Do certain answers correlate with each other?

# Summarize your inference and write a conclusion
